### PR TITLE
Added Opus stream support, added volume interface

### PR DESCRIPTION
**Pull request by Programmix**
Opus stream support is important if you want to easily stream Opus data.
The volume code for StreamDispatcher and VoiceBroadcast was almost completely identical, so I merged it into one interface.

Also did various cleanups and fixes around voice-related things.

### CODE REVIEW

File: src/client/voice/VoiceBroadcast.js [hunk 1]
```diff
@@ -336,9 +310,14 @@ class VoiceBroadcast extends EventEmitter {
     };
 
     for (const dispatcher of this.dispatchers) {
+      if (opus) {
+        dispatcher.processPacket(buffer);
+        continue;
+      }
+
       const volume = dispatcher.volume;
       setImmediate(() => {
```
Comment (hydrabolt) [inline src/client/voice/VoiceBroadcast.js]: can you remove the `setImmediate`

**Comment by PgBiel** [comment]
y u `require("events").EventEmitter` that's so redundant
just do `require("events")`
:/

**Comment by itslukej** [comment]
harder daddy

**Comment by Programmix** [comment]
Was about to fix the potential memory leak issue for the Opus decoders used in VoiceReceiver as well, but I'm waiting for PR #1132 to be merged because it relies on changes made in that PR.

**Pull request merged**